pages/search/search.jsx

```diff
@@ -51,24 +50,97 @@ export default React.createClass({
       location[`query`] = { keyword: query };
     }
 
+    // note browserHistory.push() triggers component re-render
     browserHistory.push(location);
+  },
+  clearSearch() {
+    this.refs.searchInput.setState({value: ``}, () => {
+      this.updateBrowserHistory();
+    });
+  },
+  fetchData(params = {}, entries = [], apiPageIndex = 1) {
+    if (!params.search) {
+      // if no search keyword was passed reset search page back to its initial state
+      this.setState(this.getInitialState());
+      return;
+    }
+
+    params.page = apiPageIndex;
+
+    Service.entries
+      .get(params)
+      .then((data) => {
+        entries = entries.concat(data.results);
 
+        if (data.next) {
+          // there are more matched entries in the database we need to fetch
+          this.fetchData(params, entries, apiPageIndex+1);
```

Comment:
@alanmoo 

1) I didn't realize the `count` property returned from the API is the total count of all matched entries, not just how many entries are in that particular `page`. I thought I had to add them all up to get the total of matched entries count. Can we update the property name to something not so generic? Or at least have some docs for it?

2) Not sure how big our db will grow but for since we only have ~200 entries now, making one fetch is enough to retrieve all matched entries (`page_size` is set to 996, the closest number to 1000 that's divisible by 6). Currently every time "Load more" is clicked we display 24 more entries. Users see those entries immediately after they click the button. We *could* change it so that we don't fetch all data at once when we load the page, we make new Pulse API call every time users click on "Load more". This means users will have to wait till we hear back from the call. If we decide to make this change I suggest bumping up " the number of entries in each batch" from 24 to a bigger number. Otherwise it can be quite annoying as users have to go through "click and wait" cycle for many times just to load a few hundreds of entries. /cc @kristinashu @xmatthewx I would like to know your thoughts on this 😉